src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v2/mongo/EcommerceTransactionDataProvider.kt

```diff
@@ -144,8 +144,13 @@ class EcommerceTransactionDataProvider(
                         },
                         ::Pair
                     )
-                    .map { (baseTransaction, email) ->
-                        baseTransactionToTransactionInfoDtoV1(baseTransaction, email)
+                    .zipWith(events.collectList(), ::Pair)
+                    .map { (baseTransactionEmail, events) ->
```

Comment:
variable names are one of the most difficult choices for a developer :-) maybe something like `baseTransactionAndEmailPair`?

Reply:
done